craterpy/tests/test_classes.py

```diff
@@ -97,3 +143,717 @@ def test_antimeridian_splitting(self):
+                "lon": [0.0, 20.0],
+                "radius": [1.0, 2.0],
+                "name": ["Crater A", "Crater B"],
+                "_private": [1, 2],
+            }
+        )
+        cdb = CraterDatabase(df)
+
+        geojson_str = cdb.to_geojson()
+        self.assertIsInstance(geojson_str, str)
+        self.assertIn("Crater A", geojson_str)
+        self.assertNotIn(
+            "_private", geojson_str
+        )  # Private columns should be dropped by default
+
+    def test_to_geojson_with_custom_geometry(self):
+        """Test export with a custom geometry column."""
+        df = pd.DataFrame(
+            {"lat": [0.0, 10.0], "lon": [0.0, 20.0], "radius": [1.0, 2.0]}
+        )
+        cdb = CraterDatabase(df)
+        cdb.add_circles("rim")
+
+        # Export with crater rim geometries
+        geojson_with_rim = cdb.to_geojson(geometry_column="rim")
+        self.assertIn(
+            "Polygon", geojson_with_rim
+        )  # Rims should be polygons not points
+
+    def test_to_geojson_with_specific_properties(self):
+        """Test export with specific property columns."""
+        df = pd.DataFrame(
+            {
+                "lat": [0.0, 10.0],
+                "lon": [0.0, 20.0],
+                "radius": [1.0, 2.0],
+                "name": ["Crater A", "Crater B"],
+                "description": ["Description A", "Description B"],
+            }
+        )
+        cdb = CraterDatabase(df)
+
+        # Export with only name property
+        props_only = cdb.to_geojson(properties=["name"])
+        self.assertIn("name", props_only)
+        self.assertIn("Crater A", props_only)
+        self.assertNotIn("description", props_only)
+        self.assertNotIn("Description A", props_only)
+
+    def test_to_geojson_with_private_columns(self):
+        """Test export including private columns."""
+        df = pd.DataFrame(
+            {
+                "lat": [0.0, 10.0],
+                "lon": [0.0, 20.0],
+                "radius": [1.0, 2.0],
+                "_private": [1, 2],
+                "_internal": ["x", "y"],
+            }
+        )
+        cdb = CraterDatabase(df)
+
+        # By default, private columns should be dropped
+        default_export = cdb.to_geojson()
+        self.assertNotIn("_private", default_export)
+
+        # With drop_private=False, they should be included
+        with_private = cdb.to_geojson(drop_private=False)
+        self.assertIn("_private", with_private)
+        self.assertIn("_internal", with_private)
+
+    def test_to_geojson_file_export(self):
+        """Test export to a GeoJSON file."""
+        df = pd.DataFrame(
+            {
+                "lat": [0.0, 10.0],
+                "lon": [0.0, 20.0],
+                "radius": [1.0, 2.0],
+                "name": ["Crater A", "Crater B"],
+            }
+        )
+        cdb = CraterDatabase(df)
+
+        import tempfile
+        import os
+        import json
```

Comment:
might as well move all the internal imports to the top since most are done multiple times